src/common/form/Form.js

```diff
@@ -0,0 +1,52 @@
+// @flow
+
+import * as React from 'react';
+import { Provider, observer } from 'mobx-react';
+import FormStore from './FormStore';
+
+@observer
+class Form extends React.Component {
+  formStore: FormStore;
+
+  componentWillMount() {
+    this.formStore = new FormStore({ validate: this.props.validate });
+  }
+
+  handleSubmit = async (
```

Comment:
nit: `_onSubmit`

Reply:
no, I propagate it as a prop to Form children, so it shouldn't be private. And this is how it's called in redux-forms

Reply:
Ok. I would think that `onSubmit` is a better name since it's a handler. Do whatever is easier to implement.

Reply:
I just want that it be as much similar as in redux-forms as possible.
handleSumbit is used in actual form as `handleSumbit(this.onSubmit)`. It would be weird to see `onSubmit(this.onSubmit)`